records_mover/records/mover.py

```diff
@@ -123,8 +123,9 @@ def move(records_source: RecordsSource,
                 as fileobjs_source:
             return move(fileobjs_source, records_target, processing_instructions)
     elif (isinstance(records_source, SupportsMoveToRecordsDirectory) and
-          isinstance(records_target, SupportsMoveFromTempLocAfterFillingIt) and
-          records_source.has_compatible_format(records_target)):
+          isinstance(records_target, MightSupportMoveFromTempLocAfterFillingIt) and
+          records_source.has_compatible_format(records_target) and
+          records_target.can_move_from_temp_loc_after_filling_it()):
```

Comment:
This is a key bit - I'm changing the move() function (the key algorithm for Records Mover) to only use this interface's function if the interface says it's OK.

In the most common case of a Table target, that means that the database in question has a temporary bucket location configured that it can do a bulk export to.